include/picongpu/fields/MaxwellSolver/Yee/Yee.hpp

```diff
@@ -163,6 +169,142 @@ namespace picongpu
+                }
+
+                /** Propagate B values in the given area by the second half of a time step
+                 *
+                 * This operation propagates grid values of field B by dt/2 and relies on the internal state of
+                 * convolutional components set up by a prior call to updateBFirstHalf(). After this call is
+                 * completed, the convolutional components are in the state to call updateBFirstHalf() for the
+                 * next time step.
+                 *
+                 * @tparam T_Area area to apply updates to, the curl must be applicable to all points;
+                 * normally CORE, BORDER, or CORE + BORDER
+                 *
+                 * @param currentStep index of the current time iteration
+                 */
+                template<uint32_t T_Area>
+                void updateBSecondHalf(uint32_t const currentStep)
+                {
+                    updateBHalf<T_Area>(currentStep, false);
+                }
+
+                /** Propagate B values in the given area by half a time step
+                 *
+                 * @tparam T_Area area to apply updates to, the curl must be
+                 * applicable to all points; normally CORE, BORDER, or CORE + BORDER
+                 *
+                 * @param currentStep index of the current time iteration
+                 * @param updatePsiB whether convolutional magnetic fields need to be updated, or are
+                 * up-to-date
+                 */
+                template<uint32_t T_Area>
+                void updateBHalf(uint32_t const currentStep, bool const updatePsiB)
+                {
+                    constexpr auto numWorkers = getNumWorkers();
+                    using Kernel = yee::KernelUpdateB<numWorkers, BlockDescription<CurlE>>;
+                    AreaMapper<T_Area> mapper{cellDescription};
+
+                    // The ugly transition from run-time to compile-time polymorphism is contained here
+                    auto& absorber = absorber::Absorber::get();
+                    if(absorber.getKind() == absorber::Absorber::Kind::Pml)
+                    {
+                        auto& pmlImpl = absorberImpl->asPmlImpl();
+                        auto const updateFunctor
+                            = pmlImpl.getUpdateBHalfFunctor<CurlE, T_Area>(currentStep, updatePsiB);
+                        PMACC_KERNEL(Kernel{})
+                        (mapper.getGridDim(),
+                         numWorkers)(mapper, updateFunctor, fieldE->getDeviceDataBox(), fieldB->getDeviceDataBox());
+                    }
+                    else
+                        PMACC_KERNEL(Kernel{})
+                    (mapper.getGridDim(), numWorkers)(
+                        mapper,
+                        yee::UpdateBHalfFunctor<CurlE>{},
+                        fieldE->getDeviceDataBox(),
+                        fieldB->getDeviceDataBox());
```

Comment:
@psychocoderHPC @j-stephan here is another example of our clang-formatting not managing the (admittedly, ugly) code. The resulting formatting is really illogical.

Reply:
That's right, this looks strange.

Reply:
Does this change if you put braces around the statement? I can imagine that clang-format gets confused because of the multiline statement. IIRC some compilers even issue warnings if you do this.

Reply:
Braces didn't change anything. However, `clang-format` didn't mind me somewhat fixing it manually (so both how it done it originally and how it is now pass).